bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/discovery/Clip2ThingDiscoveryService.java

```diff
@@ -0,0 +1,189 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.hue.internal.HueBindingConstants;
+import org.openhab.binding.hue.internal.dto.clip2.MetaData;
+import org.openhab.binding.hue.internal.dto.clip2.Resource;
+import org.openhab.binding.hue.internal.dto.clip2.ResourceReference;
+import org.openhab.binding.hue.internal.dto.clip2.enums.Archetype;
+import org.openhab.binding.hue.internal.dto.clip2.enums.ResourceType;
+import org.openhab.binding.hue.internal.exceptions.ApiException;
+import org.openhab.binding.hue.internal.exceptions.AssetNotLoadedException;
+import org.openhab.binding.hue.internal.handler.Clip2BridgeHandler;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+
+/**
+ * Discovery service to find resource things on a Hue Bridge that is running CLIP 2.
+ *
+ * @author Andrew Fiddian-Green - Initial Contribution
+ */
+@NonNullByDefault
+public class Clip2ThingDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+
+    public static final int DISCOVERY_TIMEOUT_SECONDS = 20;
+    public static final int DISCOVERY_INTERVAL_SECONDS = 600;
+
+    /**
+     * Map of resource types and respective thing types that shall be discovered.
+     */
+    public static final Map<ResourceType, ThingTypeUID> DISCOVERY_TYPES = Map.of( //
+            ResourceType.DEVICE, HueBindingConstants.THING_TYPE_DEVICE, //
+            ResourceType.ROOM, HueBindingConstants.THING_TYPE_ROOM, //
+            ResourceType.ZONE, HueBindingConstants.THING_TYPE_ZONE, //
+            ResourceType.BRIDGE_HOME, HueBindingConstants.THING_TYPE_ZONE);
+
+    private @Nullable Clip2BridgeHandler bridgeHandler;
+    private @Nullable ScheduledFuture<?> discoveryTask;
+
+    public Clip2ThingDiscoveryService() {
+        super(Set.of(HueBindingConstants.THING_TYPE_DEVICE, HueBindingConstants.THING_TYPE_ROOM,
+                HueBindingConstants.THING_TYPE_ZONE), DISCOVERY_TIMEOUT_SECONDS, true);
+    }
+
+    @Override
+    public void activate() {
+        Clip2BridgeHandler bridgeHandler = this.bridgeHandler;
+        if (Objects.nonNull(bridgeHandler)) {
+            bridgeHandler.registerDiscoveryService(this);
+        }
+        super.activate(null);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+        Clip2BridgeHandler bridgeHandler = this.bridgeHandler;
+        if (Objects.nonNull(bridgeHandler)) {
+            bridgeHandler.registerDiscoveryService(null);
+            removeOlderResults(new Date().getTime(), bridgeHandler.getThing().getBridgeUID());
```

Comment:
You could also set `this.bridgeHandler` to null.

Reply:
Ok. Done.